internal/manifests/collector/horizontalpodautoscaler.go

```diff
@@ -48,6 +46,7 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 		logger.Info("Autoscaler field is unset in Spec, skipping")
 		return nil
 	}
+	autoscalingVersion := cfg.AutoscalingVersion()
```

Comment:
nit, but why did we move this here? I did expect to keep it where it is, or moving the part where its actually used. ^^